### PR TITLE
The mvps first sprint

### PR DESCRIPTION
**Pull request by olivia-k-mason**
Created ProductRepository, has dependencies on ApiProduct and Product data class

Also created Name model and Category model

### CODE REVIEW

**Review by kohlmank — APPROVED** [review]
Looks Good!

File: app/src/main/java/com/capgemini/menuapp/model/Category.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.capgemini.menuapp.model
+import com.google.gson.annotations.SerializedName
+
+data class Category(
+  @SerializedName("CategoryID")
+  val category: Int = 0,
```
Comment (Dave-Murdoch) [inline app/src/main/java/com/capgemini/menuapp/model/Category.kt:6]: id

File: app/src/main/java/com/capgemini/menuapp/repository/ProductRepository.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.capgemini.menuapp.repository
+import io.reactivex.Single
+
+
+class ProductRepository(private val apiProduct: ApiProduct){
```
Comment (Dave-Murdoch) [inline app/src/main/java/com/capgemini/menuapp/repository/ProductRepository.kt:5]: Takes a ProductService

File: app/src/main/java/com/capgemini/menuapp/repository/ProductRepository.kt [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+  fun getProduct(): Single<List<Product>>
```
Comment (Dave-Murdoch) [inline app/src/main/java/com/capgemini/menuapp/repository/ProductRepository.kt:6]: getProducts()

File: app/src/main/java/com/capgemini/menuapp/repository/ProductRepository.kt [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+        return apiProduct.getProduct()
```
Comment (Dave-Murdoch) [inline app/src/main/java/com/capgemini/menuapp/repository/ProductRepository.kt:7]: See above comments

**Pull request merged**